test/e2e/clusterclass_rollout.go

```diff
@@ -672,43 +672,32 @@ func assertMachinePools(g Gomega, clusterClassObjects clusterClassObjects, clust
 		))
 
 		// MachinePool InfrastructureMachineTemplate.metadata
-		ccInfrastructureMachineTemplate := clusterClassObjects.InfrastructureMachinePoolTemplateByMachinePoolClass[mpClass.Class]
-		ccInfrastructureMachineTemplateTemplateMetadata := mustMetadata(contract.InfrastructureMachineTemplate().Template().Metadata().Get(ccInfrastructureMachineTemplate))
-		infrastructureMachineTemplate := clusterObjects.InfrastructureMachinePoolTemplateByMachinePool[machinePool.Name]
-		infrastructureMachineTemplateTemplateMetadata := mustMetadata(contract.InfrastructureMachineTemplate().Template().Metadata().Get(infrastructureMachineTemplate))
-		g.Expect(infrastructureMachineTemplate.GetLabels()).To(BeEquivalentTo(
+		ccInfrastructureMachinePoolTemplate := clusterClassObjects.InfrastructureMachinePoolTemplateByMachinePoolClass[mpClass.Class]
+		infrastructureMachinePool := clusterObjects.InfrastructureMachinePoolByMachinePool[machinePool.Name]
+		g.Expect(infrastructureMachinePool.GetLabels()).To(BeEquivalentTo(
 			union(
 				map[string]string{
 					clusterv1.ClusterNameLabel:                    cluster.Name,
 					clusterv1.ClusterTopologyOwnedLabel:           "",
 					clusterv1.ClusterTopologyMachinePoolNameLabel: mpTopology.Name,
 				},
-				ccInfrastructureMachineTemplate.GetLabels(),
+				ccInfrastructureMachinePoolTemplate.GetLabels(),
```

Comment:
```suggestion
				ccInfrastructureMachinePoolTemplateTemplateMetadata.GetLabels(),
```

I think we take the labels from (e.g.) DockerMachinePoolTemplate.spec.template.metadata.labels and use them then for DockerMachinePool.metadata.labels

The new var can be calculate like that:
```go
ccInfrastructureMachinePoolTemplateTemplateMetadata := mustMetadata(contract.InfrastructureMachinePoolTemplate().Template().Metadata().Get(ccInfrastructureMachinePoolTemplate))
```
(we also have to create the InfrastructureMachinePoolTemplate func in the contract package)